container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java

```diff
@@ -50,4 +58,49 @@ public Model modelForRankProfile(String rankProfile) {
     public FunctionEvaluator evaluatorForFunction(String rankProfile, String functionName) {
         return modelForRankProfile(rankProfile).evaluatorOf(functionName);
     }
+
+    static record GlobalPhaseData(Supplier<FunctionEvaluator> functionEvaluatorSource,
+                                  int rerankCount,
+                                  List<String> needInputs) {}
+
+    private Map<String, GlobalPhaseData> profilesWithGlobalPhase = new HashMap<>();
+
+    GlobalPhaseData getGlobalPhaseData(String rankProfile) {
+        return profilesWithGlobalPhase.get(rankProfile);
+    }
+
+    private void extractGlobalPhaseData(RankProfilesConfig rankProfilesConfig) {
+        for (var rp : rankProfilesConfig.rankprofile()) {
+            String name = rp.name();
+            Supplier<FunctionEvaluator> functionEvaluatorSource = null;
+            int rerankCount = -1;
+            List<String> needInputs = null;
+
+            for (var prop : rp.fef().property()) {
+                if (prop.name().equals("vespa.globalphase.rerankcount")) {
+                    try {
+                        rerankCount = Integer.valueOf(prop.value());
+                    } catch (NumberFormatException e) {
+                        logger.warning("bad vespa.globalphase.rerankcount '" + prop.value() +
```

Comment:
It's a safe assumption to fail the component (graph) construction if config is invalid.